components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -29,20 +29,27 @@ fn json_array(args: &[&Option<Json>]) -> Result<Option<Json>> {
     )))
 }
 
-/// `&[&(Option<Byte>, Option<Json>)]`
-/// TODO: if we need to capture Expression to handle the error here?
-/// TODO: add errors for key is None
-#[rpn_fn(raw_varg, extra_validator = json_object_validator)]
+/// Required args like `&[&(Option<Byte>, Option<Json>)]`.
+#[rpn_fn(raw_varg)]
 #[inline]
 fn json_object(raw_args: &[ScalarValueRef]) -> Result<Option<Json>> {
     let mut pairs = BTreeMap::new();
     for chunk in raw_args.chunks(2) {
+        if chunk.len() == 1 {
+            return Err(other_err!(
+                "Incorrect parameter count in the call to native function 'JSON_OBJECT'"
+            ));
+        }
         let key: &Option<Bytes> = Evaluable::borrow_scalar_value_ref(&chunk[0]);
         let key = match key {
             // json_object should raise an error if key is None(NULL)
-            None => unimplemented!(),
-            Some(v) => String::from_utf8(v.to_owned()).map_err(|e| crate::codec::Error::from(e))?,
-        };
+            None => Err(other_err!(
+                "Data truncation: JSON documents may not contain NULL member names."
+            )),
+            Some(v) => {
+                String::from_utf8(v.to_owned()).map_err(|e| crate::codec::Error::from(e).into())
```

Comment:
Could we omit `.map_err(|e| crate::codec::Error::from(e).into())`?

Reply:
> Could we omit `.map_err(|e| crate::codec::Error::from(e).into())`?

This expression returns `tidb_query::Error`, do you have any idea to omit it?